### PR TITLE
Fixed a redeem bug and added more Homa lite tests

### PR DESCRIPTION
**Pull request by syan095**
Fixed a bug where when redeeming with available staking balance the staking total is not updated
Refactored the code so all redeem logic shares the same code
Added a new mock that have no fees to test economic model
Added tests to ensure that when no fees are charged, the exchange rate says the same throughout all operations.
Modified `RedeemRequested` event to include the fee being charged.

### CODE REVIEW

File: modules/homa-lite/src/lib.rs [hunk 1]
```diff
@@ -488,74 +492,42 @@ pub mod module {
...
-				// Unredeemed requests are added to a queue.
-				if Self::liquid_amount_is_above_minimum_threshold(liquid_remaining) {
-					// Check if there's already a queued redeem request.
-					let (request_amount, _) = Self::redeem_requests(&who).unwrap_or((0, Permill::default()));
-
-					match liquid_remaining.cmp(&request_amount) {
-						// Lock more liquid currency.
-						Ordering::Greater => T::Currency::reserve(
+						liquid_amount.saturating_sub(old_amount),
+					),
```
Comment (xlc) [inline modules/homa-lite/src/lib.rs:502]: why not just error return here?
```suggestion
					)?,
```

File: modules/homa-lite/src/lib.rs [hunk 2]
```diff
@@ -473,7 +476,8 @@ pub mod module {
 				Error::<T>::AmountBelowMinimumThreshold
 			);
 
-			RedeemRequests::<T>::try_mutate(&who, |request| -> DispatchResult {
+			// Deduct base withdraw fee and add the redeem request to the queue.
+			let _ = RedeemRequests::<T>::try_mutate(&who, |request| -> DispatchResult {
```
Comment (xlc) [inline modules/homa-lite/src/lib.rs:480]: why ignore error?

File: modules/homa-lite/src/lib.rs [hunk 3]
```diff
@@ -488,74 +492,42 @@ pub mod module {
...
+					additional_fee,
+					base_withdraw_fee,
+				));
 
 				Ok(())
-			})
+			});
+
+			// With redeem request added to the queue, try to redeem it with available staking balance.
+			let _ = Self::process_redeem_requests_with_available_staking_balance(1)?;
```
Comment (xlc) [inline modules/homa-lite/src/lib.rs:529]: why ignore result?

File: modules/homa-lite/src/lib.rs (hunk 1 above)
Comment (syan095) [reply modules/homa-lite/src/lib.rs:502]: true. I will add more test to ensure this will fail when the user has insufficent fund

**Review by wangjj9219 — APPROVED** [review]

**Pull request merged**